app/presenters/curation_concerns/inspect_work_presenter.rb

```diff
@@ -0,0 +1,66 @@
+module CurationConcerns
+  class InspectWorkPresenter
+    def initialize(solr_document, current_ability)
+      @solr_document = solr_document
+      @current_ability = current_ability
+    end
+    attr_reader :solr_document, :current_ability
+
+    def workflow
+      workflow_resp = { comments: [], roles: [] }
+      if sipity_entity
+        workflow_resp =
+          {
+            entity_id: sipity_entity.id,
+            proxy_for: sipity_entity.proxy_for_global_id,
+            workflow_id: sipity_entity.workflow_id,
+            workflow_name: sipity_entity.workflow_name,
+            state_id: sipity_entity.workflow_state_id,
+            state_name: sipity_entity.workflow_state_name,
+            comments: workflow_comments,
+            roles: sipity_entity_roles
+          }
+      end
+      workflow_resp
+    end
+
+    def solr
+      @solr_document.inspect
+    end
+
+    private
+
+      def sipity_entity
+        @sipity_entity ||= PowerConverter.convert(solr_document, to: :sipity_entity)
+      rescue PowerConverter::ConversionError
+        nil
+      end
+
+      def sipity_entity_roles
+        @roles = CurationConcerns::Workflow::PermissionQuery.scope_roles_associated_with_the_given_entity(entity: solr_document)
+        roles = []
+        @roles.each do |role|
+          roles << { id: role.id, name: role.name, description: role.description, users: sipity_entity_role_users(role) }
+        end
+        roles
+      end
+
+      def sipity_entity_role_users(role)
+        @users = CurationConcerns::Workflow::PermissionQuery.scope_users_for_entity_and_roles(entity: sipity_entity, roles: role)
+        users = []
```

Comment:
These 5 lines could be changed to be: `@users.map { |user| user.user_key }` It's preferable to use `user_key`, because that field can be customized although it is `email` by default. Some institutions use `username`